google/cloud/internal/populate_common_options_test.cc

```diff
@@ -49,6 +49,62 @@ TEST(PopulateCommonOptions, Simple) {
               Contains(UserAgentPrefix()));
 }
 
+TEST(PopulateCommonOptions, EmptyEndpointOption) {
+  // Unset all the relevant environment variables.
+  ScopedEnvironment user("GOOGLE_CLOUD_CPP_USER_PROJECT", absl::nullopt);
+  ScopedEnvironment tracing("GOOGLE_CLOUD_CPP_ENABLE_TRACING", absl::nullopt);
+  auto actual = PopulateCommonOptions(Options{}.set<EndpointOption>(""), {}, {},
+                                      {}, "default");
+  EXPECT_TRUE(actual.has<EndpointOption>());
+  EXPECT_THAT(actual.get<EndpointOption>(), Eq("default"));
+  EXPECT_TRUE(actual.has<AuthorityOption>());
+  EXPECT_THAT(actual.get<AuthorityOption>(), Eq("default"));
+  EXPECT_FALSE(actual.has<UserProjectOption>());
+  EXPECT_TRUE(actual.has<TracingComponentsOption>());
+  EXPECT_THAT(actual.get<TracingComponentsOption>(), IsEmpty());
+  EXPECT_TRUE(actual.has<UserAgentProductsOption>());
+  EXPECT_THAT(actual.get<UserAgentProductsOption>(),
+              Contains(UserAgentPrefix()));
+}
+
+TEST(PopulateCommonOptions, EmptyEndpointEnvVar) {
```

Comment:
Actually, I think we can remove this test. I think it is covered by `EndpointAuthority` (not that I love that test. It has very complicated logic for a unit test).